### PR TITLE
Adding showing email of the author of deck

### PR DESCRIPTION
**Pull request by ArekArek**
Now it's not connected with db.
And changed GET in decks resources to return deckrepresentation

### CODE REVIEW

File: backend/app/src/main/java/com/bls/patronage/resources/DecksResource.java [hunk 1]
```diff
@@ -155,4 +159,17 @@ public DeckCollectionBuilder getRandom(Optional<Boolean> random) {
             return decks;
         }
     }
+
+    private Collection<DeckRepresentation> deckCollectionToDeckRespresentationCollection(Collection<Deck> deckCollection) {
+        Collection<DeckRepresentation> deckRepresentationCollection;
```
Comment (mlotysz) [inline backend/app/src/main/java/com/bls/patronage/resources/DecksResource.java]: final missing

File: backend/app/src/main/java/com/bls/patronage/resources/DecksResource.java [hunk 2]
```diff
@@ -155,4 +159,17 @@ public DeckCollectionBuilder getRandom(Optional<Boolean> random) {
...
+        List deckRepresentations = new ArrayList<>();
```
Comment (mlotysz) [inline backend/app/src/main/java/com/bls/patronage/resources/DecksResource.java]: final missing

File: backend/app/src/main/java/com/bls/patronage/resources/DecksResource.java [hunk 3]
```diff
@@ -155,4 +159,17 @@ public DeckCollectionBuilder getRandom(Optional<Boolean> random) {
...
+        Deck deck;
```
Comment (mlotysz) [inline backend/app/src/main/java/com/bls/patronage/resources/DecksResource.java]: final missing

File: backend/app/src/main/java/com/bls/patronage/resources/DecksResource.java [hunk 4]
```diff
@@ -155,4 +159,17 @@ public DeckCollectionBuilder getRandom(Optional<Boolean> random) {
...
+
+        for (Iterator<Deck> i =deckCollection.iterator(); i.hasNext(); ){
```
Comment (mlotysz) [inline backend/app/src/main/java/com/bls/patronage/resources/DecksResource.java]: formatting

File: backend/app/src/main/java/com/bls/patronage/resources/DecksResource.java [hunk 5]
```diff
@@ -155,4 +159,17 @@ public DeckCollectionBuilder getRandom(Optional<Boolean> random) {
...
+            deck=i.next();
+            deckRepresentations.add(new DeckRepresentation(deck).setCreatorEmail(decksDAO.getCreatorEmailFromDeckId(deck.getId())));}
+        deckRepresentationCollection=deckRepresentations;
```
Comment (mlotysz) [inline backend/app/src/main/java/com/bls/patronage/resources/DecksResource.java]: return deckRepresentations?

Comment (ArekArek) [inline backend/app/src/main/java/com/bls/patronage/resources/DecksResource.java]: Yes, to be independent from db model

Comment (mlotysz) [inline backend/app/src/main/java/com/bls/patronage/resources/DecksResource.java]: deckRepresentationCollection is not needed you can do `return deckRepresentations;` instead.

File: backend/app/src/main/java/com/bls/patronage/resources/DecksResource.java [hunk 6]
```diff
@@ -155,4 +160,15 @@ public DeckCollectionBuilder getRandom(Optional<Boolean> random) {
             return decks;
         }
     }
+
+    private Collection<DeckRepresentation> deckCollectionToDeckRespresentationCollection(Collection<Deck> deckCollection) {
+        final List deckRepresentations = new ArrayList<>();
+        Deck deck;
```
Comment (mlotysz) [inline backend/app/src/main/java/com/bls/patronage/resources/DecksResource.java]: Remove this - it's temp value.

File: backend/app/src/main/java/com/bls/patronage/resources/DecksResource.java [hunk 7]
```diff
@@ -155,4 +160,15 @@ public DeckCollectionBuilder getRandom(Optional<Boolean> random) {
...
+
+        for (Iterator<Deck> i = deckCollection.iterator(); i.hasNext(); ) {
+            deck = i.next();
```
Comment (mlotysz) [inline backend/app/src/main/java/com/bls/patronage/resources/DecksResource.java]: Inline this.

File: backend/app/src/main/java/com/bls/patronage/resources/DecksResource.java [hunk 8]
```diff
@@ -155,4 +160,15 @@ public DeckCollectionBuilder getRandom(Optional<Boolean> random) {
             return decks;
         }
     }
+
+    private Collection<DeckRepresentation> deckCollectionToDeckRespresentationCollection(Collection<Deck> deckCollection) {
+        final List deckRepresentations = new ArrayList<>();
+        Deck deck;
+
+        for (Iterator<Deck> i = deckCollection.iterator(); i.hasNext(); ) {
```
Comment (mlotysz) [inline backend/app/src/main/java/com/bls/patronage/resources/DecksResource.java]: Use `deckcollection.stream.map`. Iterators are fine when you modify some collection when iterating... This is too much code for what is should be.

**Comment by mlotysz** [comment]
Thanks!

**Comment by mlotysz** [comment]
Let's optimize for each next time.

**Pull request merged**